### PR TITLE
[proposal] propose new instructions in kubefile

### PR DESCRIPTION
**Pull request by justadogistaken**
### Describe what this PR does / why we need it
related to [[Proposal] A more k8s-based/rootfs-transparent instructions in Kubefile](https://github.com/sealerio/sealer/issues/1579)

### Special notes for reviews

For CNI/CSI/APP, I still fail to think throgh the dynamic render for remote fetching(https) files at runtime (this is easy to accomplish at local). 

This is note a ultimate version, propose this to have a discussion with you guys. @sealerio/sealer-maintainers @sealerio/sealer-reviewers

### CODE REVIEW

**Comment by kakaZhou719** [comment]
@justadogistaken ,

Very excited to see our newly refactored kubefile instruction, i have some questions about the `LANGCH` instruction and offline delivery.

1. Different applications start in different ways, such as: helm package, yaml file application. Do we need to distinguish them by adding several parameters?
2. Could we support building applications with variable parameters? such as the image name in yaml which can be specified at run stage, build once and run it according to different parameters.
3. Some applications are started by the operator, after that the related pods are started. if cluster image is delivered to the offline environment, the images of those pods need to be saved in clusterimage in advance.

**Comment by justadogistaken** [comment]
@kakaZhou719 
Thanks for your suggestions.
1. I agree with you, we will support options/flags to support various app installation.
2. For this problem, how do we handle this at present? I guess it dues to env rending?(I' not so sure), and of course I can allow users to input some args at the `LAUNCH`.
3. I'm confused with this problem. In this way, it won't affect image saving. What you mean is `COPY imageList manifest`? We won't remove the original instructions, and I can also provide a instruction to support that. And is this related to `LAUNCH`, I still feel confused. Maybe you can share more detailed information.

File: docs/design/kubefile_alpha.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+base image could be any of cluster image or scratch.
+
+### CNI
+
+`CNI` allows user to define which cni to be installed over Kubernetes.
+
+Some behaviors are available:
+
+`CNI --type calico / flannel --cidr ...` a easiest way to deploy cni, sealer will support several types of cni.
```
Comment (allencloud) [inline docs/design/kubefile_alpha.md:75]: Please make all CNI and CSI uppercase, and search them globally in this file and update them all.

s/a easiest way/an easiest way/g

**Comment by allencloud** [comment]
./docs/design/kubefile_alpha.md:7: MD009 Trailing spaces
./docs/design/kubefile_alpha.md:8: MD009 Trailing spaces
./docs/design/kubefile_alpha.md:9: MD009 Trailing spaces
./docs/design/kubefile_alpha.md:110: MD012 Multiple consecutive blank lines
./docs/design/kubefile_alpha.md:50: MD031 Fenced code blocks should be surrounded by blank lines
./docs/design/kubefile_alpha.md:5[6](https://github.com/sealerio/sealer/runs/7461921658?check_suite_focus=true#step:5:7): MD031 Fenced code blocks should be surrounded by blank lines
./docs/design/kubefile_alpha.md:[8](https://github.com/sealerio/sealer/runs/7461921658?check_suite_focus=true#step:5:9)8: MD038 Spaces inside code span elements

File: docs/design/kubefile_alpha.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+`CNI --source [path / https://...]`. `path` is a relative path refers to local build context, this can be a directory
+contains multiple yaml or a single yaml for deploying cni. `https://...` is a remote address for downloading related files to deploy cni.
+
+`[]` means user could declare several sources
```
Comment (allencloud) [inline docs/design/kubefile_alpha.md:80]: I think we could only declare CNI once. So we could not support [] which means multiple sources.

File: docs/design/kubefile_alpha.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+### CSI
+
+`CSI` allows user to define which csi to be installed over Kubernetes.
+
+Some behaviors are available:
+
+`CSI --type alibaba-cloud-csi-driver ` a easiest way to deploy csi, sealer will support several types of csi.
+
+`CSI --source [path / https://...]`. `path` is a relative path refers to local build context, this can be a directory
```
Comment (allencloud) [inline docs/design/kubefile_alpha.md:90]: I think we can remove all `--source` flag. This is redundant, I think. 
We can just judge which kind of source by the input value.

Less flag could reduce more mind burden for Kubefile end users.

File: docs/design/kubefile_alpha.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+Some behaviors are available:
+
+`APP [UNIQUE-NAME] --source [path / https://...]`. `[UNIQUE-NAME]` is a declaration for an application, which is unique. `path` is a relative path refers to local build context, this can be a directory
+contains multiple yaml or a single file for deploying app. `https://...` is a remote address for downloading related files to deploy app.
+
+`[]` means user could declare several sources
+
+### LAUNCH
+
+`LAUNCH` allows user to specify which apps to start right after the completion of cluster initing.
```
Comment (allencloud) [inline docs/design/kubefile_alpha.md:108]: s/initing/initialization

**Review by allencloud — APPROVED** [review]

**Comment by allencloud** [comment]
LGTM

**Pull request merged**